client-js/app/elements/labrad-nodes.ts

```diff
@@ -626,47 +636,43 @@ export class LabradNodes extends polymer.Base {
   }
 
 
-  private updateServerAutostartStatus(serversName: string,
-                                      serverIdx: number): void {
-    const server = this[serversName][serverIdx];
-    server.autostart = false;
-    for (const node of server.nodes) {
-      if (node.exists && node.autostart) {
-        server.autostart = true;
-        return;
+  private updateServerVersion(listName: string, serverIdx: number): void {
+    const server = this[listName][serverIdx];
+    for (let nodeIdx = 0; nodeIdx < server.nodes.length; nodeIdx++) {
+      const node = server.nodes[nodeIdx];
+      if (node.exists) {
+        const status = this.getServerStatus(node.name, server.name);
+        const version = status.outdated
+                      ? `running=${status.runningVersion}, latest=${status.latestVersion}`
+                      : status.version;
+        if (node.version !== version) {
+          this.set(`${listName}.#${serverIdx}.nodes.#${nodeIdx}.version`,
+                   version);
```

Comment:
Unfortunate is an understatement :-)

What I really want here is to use react, so we can just build up a complete model of the state of all the nodes after every event and then tell it to re-render the UI, and let _it_ figure out the minimal set of DOM changes that are needed to get the UI in sync. Here, we have to redirect all the state changes through polymer's `set` function with this complicated path syntax so that the polymer data binding infrastructure is aware of the changes we are making. It's really gross.

Incidentally, I looked into what it would take to drop react in here and it would require a bunch of changes to the build system to get that working, but stay tuned for future PRs :-)